slu/slu/dev/prompt_setup.py

```diff
@@ -0,0 +1,269 @@
+    REGEX3 = re.compile("/\W/g")
+    REGEX4 = re.compile(r"{{(.*?)}}")
+
+    if not isinstance(prompt, str):
+        return prompt
+
+    prompt = prompt.lower()
+    prompt = re.sub(REGEX1, "", prompt)
+    prompt = re.sub(REGEX2, "", prompt)
+    prompt = re.sub(REGEX3, "", prompt)
+    prompt = re.sub(" +", " ", prompt)
+    prompt = re.sub("_", " ", prompt)
+    prompt = prompt.strip()
+    if remove_var:
+        prompt = re.sub(REGEX4, fill_token, prompt)
+    prompt = prompt.translate(
+        str.maketrans("", "", string.punctuation.replace("<", "").replace(">", ""))
+    )
+
+    return prompt
+
+
+def _valid_string(string: str) -> bool:
+    if isinstance(string, str):
+        if all(
+            [
+                len(string) > 0,
+                string != "nan",
+                string != ".nan",
+                string != "",
+                string != " ",
+                "Unnamed" not in string,
+            ]
+        ):
+            return True
+    return False
+
+
+def _nls_to_state(string: str, delimiter: str = "_") -> str:
+    """
+    Convert NLS-Key to State.
+    NLS-Keys are nothing but extentions of State names.
+    Eg: state "A" will have NLS-Keys names A_1, A_2, and so on.
+    This code simply removes the suffix _1, _2, etc from a NLS-Key to get the original State name.
+    """
+    if not _valid_string(string):
+        return None
+    string = string.split(delimiter)
+    if len(string) > 1:
+        string = str(delimiter).join(_ for _ in string[: len(string) - 1])
+    if isinstance(string, list) and len(string) > 0:
+        string = string[0]
+
+    return string
+
+
+def _nls_to_df(dataset: str, config: Config) -> pd.DataFrame:
+    nls_labels = None
+    nls_keys = set()
+
+    if not dataset.endswith(".yaml"):
+        raise RuntimeError(
+            f"""
+            Invalid extension, .yaml file expected but instead received {dataset}.
+            """.strip()
+        )
+
+    with open(dataset) as file:
+        nls_labels = yaml.load(file, Loader=yaml.FullLoader)
+    if not nls_labels:
+        raise RuntimeError(
+            f"""
+            Invalid or empty input file {dataset}.
+            """.strip()
+        )
+
+    for lang in config.get_supported_languages():
+        if const.NLS_LANG_MAPPING[lang] not in nls_labels:
+            raise Exception(
+                f"No prompts found for {lang}, please check your input file."
+            )
+
+        else:
+            for _ in nls_labels[const.NLS_LANG_MAPPING[lang]].keys():
+                nls_keys.add(_)
+
+    logger.debug(f"Total unique nls-keys: {len(nls_keys)}")
+
+    nls_df = pd.DataFrame(
+        columns=[const.NLS_LABEL] + list(config.get_supported_languages())
+    )
+    nls_df[const.NLS_LABEL] = pd.Series(list(nls_keys))
+
+    for i in tqdm(range(nls_df.shape[0]), desc="Fetching prompts"):
+        NLS_LABEL = nls_df.iloc[i][const.NLS_LABEL]
```

Comment:
Please use a different name for `NLS_LABEL`. You have a variable with the same name in constants also which makes the code super hard to read over here.